### PR TITLE
chore: add order by clause in search

### PR DESCRIPTION
**Pull request by arriqaaq**

### CODE REVIEW

File: pkg/api/proto/documents.proto [hunk 1]
```diff
@@ -159,11 +159,16 @@ message DocumentSearchRequest {
...
   string collection = 2;
   Query query = 3;
-  bool desc = 4;
+  OrderExpression orderBy = 4;
   uint32 page = 5;
   uint32 perPage = 6;
 }
 
+message OrderExpression {
+  string field = 1;
```
[the remark this answers is not in the document]
Comment (arriqaaq) [reply pkg/api/proto/documents.proto:168]: Currently they are not, 

`
func (stmt *SelectStmt) execAt(ctx context.Context, tx *SQLTx, params map[string]interface{}) (*SQLTx, error) {
	if stmt.groupBy == nil && stmt.having != nil {
		return nil, ErrHavingClauseRequiresGroupClause
	}

	if len(stmt.groupBy) > 1 {
		return nil, ErrLimitedGroupBy
	}

	if len(stmt.orderBy) > 1 {
		return nil, ErrLimitedOrderBy
	}
`

File: embedded/document/engine.go [hunk 2]
```diff
@@ -978,3 +986,15 @@ func (e *Engine) DeleteDocument(ctx context.Context, collectionName string, quer
 func (e *Engine) CopyCatalogToTx(ctx context.Context, tx *store.OngoingTx) error {
 	return e.sqlEngine.CopyCatalogToTx(ctx, tx)
 }
+
+func generateOrderByExpression(table *sql.Table, orderBy []*protomodel.OrderExpression) (ordCols []*sql.OrdCol) {
+	if orderBy == nil || len(orderBy) == 0 {
```
Comment (jeroiraz) [inline embedded/document/engine.go:991]: not needed...

File: embedded/document/engine_test.go [hunk 3]
```diff
@@ -1058,3 +1057,75 @@ func TestGetCollection(t *testing.T) {
...
+	)
+	require.NoError(t, err)
+
+	noOfDocs := 5
+	// add documents to collection
+	for i := 1; i <= noOfDocs; i++ {
+		_, _, err = engine.InsertDocument(context.Background(), collectionName, &structpb.Struct{
+			Fields: map[string]*structpb.Value{
+				"number": {
+					Kind: &structpb.Value_NumberValue{NumberValue: float64(i)},
```
Comment (jeroiraz) [inline embedded/document/engine_test.go:1086]: structpb.NewNumberValue

File: embedded/document/engine_test.go [hunk 4]
```diff
@@ -1058,3 +1057,75 @@ func TestGetCollection(t *testing.T) {
...
+				},
+				"age": {
+					Kind: &structpb.Value_NumberValue{NumberValue: float64(noOfDocs - i + 1)},
```
Comment (jeroiraz) [inline embedded/document/engine_test.go:1089]: structpb.NewNumberValue

File: embedded/document/engine_test.go [hunk 5]
```diff
@@ -1058,3 +1057,75 @@ func TestGetCollection(t *testing.T) {
...
+
+	query := &protomodel.Query{
+		Expressions: []*protomodel.QueryExpression{
+			{
+				FieldComparisons: []*protomodel.FieldComparison{
+					{
+						Field:    "number",
+						Operator: protomodel.ComparisonOperator(sql.GT), // EQ
+						Value: &structpb.Value{
+							Kind: &structpb.Value_NumberValue{NumberValue: 0},
```
Comment (jeroiraz) [inline embedded/document/engine_test.go:1104]: structpb.NewNumberValue

File: embedded/document/engine_test.go [hunk 6]
```diff
@@ -1058,3 +1057,75 @@ func TestGetCollection(t *testing.T) {
...
+		orderBy := []*protomodel.OrderExpression{{
+			Field: "number",
+			Desc:  true,
+		}}
+
+		reader, err := engine.GetDocuments(ctx, collectionName, query, 0, orderBy)
+		require.NoError(t, err)
+		defer reader.Close()
+		docs, err := reader.ReadN(ctx, noOfDocs)
+		// require.NoError(t, err)
```
Comment (jeroiraz) [inline embedded/document/engine_test.go:1122]: better to keep error checking

File: embedded/document/engine_test.go [hunk 7]
```diff
@@ -1058,3 +1057,75 @@ func TestGetCollection(t *testing.T) {
...
+		require.Equal(t, 5, len(docs))
```
Comment (jeroiraz) [inline embedded/document/engine_test.go:1123]: require.Len

File: pkg/database/document_database.go [hunk 8]
```diff
@@ -303,7 +303,7 @@ func (d *db) SearchDocuments(ctx context.Context, req *protomodel.DocumentSearch
 		return nil, fmt.Errorf("invalid offset")
 	}
 
-	reader, err := d.documentEngine.GetDocuments(ctx, req.Collection, req.Query, int64(offset))
+	reader, err := d.documentEngine.GetDocuments(ctx, req.Collection, req.Query, int64(offset), req.OrderBy)
```
Comment (jeroiraz) [inline pkg/database/document_database.go:306]: minor detail, but I'd put orderBy before the offset so to have the same ordering as in sql

**Pull request merged**